modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/numeric/NumericDDMFormFieldValueLocalizer.java

```diff
@@ -73,6 +74,15 @@ public String localize(String value, Locale locale) {
 
 					formattedNumber = formattedNumber.concat(lastChar);
 				}
+				else if (!NumberUtil.hasDecimalSeparator(
+							locale, formattedNumber) &&
```

Comment:
Hey, @marcosapmf. Here, the `formattedNumber` is from US. Why do you check if the US formatted number has a decimal separator specific from another locale?

Reply:
My bad, @rodrigopaulino. The right else if block should be:

```
!NumberUtil.hasDecimalSeparator(
	LocaleUtil.US, formattedNumber) &&
	NumberUtil.hasDecimalSeparator(locale, value) &&
	!_endsWithDecimalSeparator(value)
```

this scenario is needed in cases where we a decimal field with a whole integer like `1.0`. Without this code, `formattedNumber` would be just `1` because of the formatter